cogs/reminder_cog.py

```diff
@@ -0,0 +1,159 @@
+import disnake
+from disnake.ext import commands, tasks
+import dateparser
+from datetime import datetime
+import json
+import logging
+import pytz
+
+class ReminderCog(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.REMINDERS_FILE = 'reminders.json'
+        self.reminders = {}
+        self.utc = pytz.UTC
+        self.pst = pytz.timezone('America/Los_Angeles')
+        self.load_reminders()
+        self.check_reminders.start()
+
+    def load_reminders(self):
+        try:
+            with open(self.REMINDERS_FILE, 'r') as f:
+                self.reminders = json.load(f)
+        except FileNotFoundError:
+            self.reminders = {}
+
+    def save_reminders(self):
+        with open(self.REMINDERS_FILE, 'w') as f:
+            json.dump(self.reminders, f)
+
+    @tasks.loop(seconds=30)
```

Comment:
do you know if there are any performance difference between doing this every 30 secs vs every 1 min? if so we can potentially do every min instead of every 30 secs, i see 1 min being a good interval